### ISSUE TITLE
Installing VSIX extension file from command line is significantly slower

### ISSUE THREAD

**Issue by joeywatts** [microsoft/vscode#89204]
- VSCode Version: 1.41.1
- OS Version: macOS 10.14.5

Steps to Reproduce:

1. Run `code --install-extension myext.vsix`

Installing a 1.7MB VSIX file from the command line takes ~27 seconds using the command line interface, but succeeds instantly when installing via the UI.

Does this issue occur when all extensions are disabled?: N/A

**Comment by joeywatts** [comment]
After some investigation, logs revealed 26.645 seconds are spent querying https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery (the `ExtensionGalleryService`).

**Comment by gjsjohnmurray** [comment]
Does the info at https://code.visualstudio.com/docs/setup/network help? Maybe add command line parameters to force use of your proxy.

**Comment by joeywatts** [comment]
@gjsjohnmurray that seems to claim that the CLI does not support proxies at all, but I can see the code attempting to use my proxy (picks up both the http.proxy in VSCode settings and http(s)_proxy environment variables) with the http(s)-proxy-agent. I confirmed this by setting breakpoints in the code that sets up the request and the proxy agent.

**Comment by joeywatts** [comment]
After further investigation, it seems there was something messed up in my proxy configuration! I was specifying "http://localhost:8888" for my proxy and that was working fine for other programs (including npm and making requests with curl), but not VSCode. I noticed that the timeout error said 127.0.0.1, so I tried using "http://127.0.0.1:8888" and was able to reproduce it with curl.

I was able to fix it by modifying my tinyproxy config to add "Listen 127.0.0.1". Not sure why this would matter (since localhost maps to 127.0.0.1 in my /etc/hosts) or how the host seems to get lost in the mix only with VSCode. This also revealed that telemetry seems to cause it to hang after installing the extension (can easily work around this for now by just disabling telemetry).
